omniscidb/QueryBuilder/QueryBuilder.cpp

```diff
@@ -0,0 +1,3072 @@
+}
+
+BuilderNode BuilderNode::sort(const std::vector<std::string>& col_names,
+                              const std::string& dir,
+                              const std::string& null_pos,
+                              size_t limit,
+                              size_t offset) const {
+  return sort(toSortFields(col_names, dir, null_pos), limit, offset);
+}
+
+BuilderNode BuilderNode::sort(BuilderExpr col_ref,
+                              SortDirection dir,
+                              NullSortedPosition null_pos,
+                              size_t limit,
+                              size_t offset) const {
+  return sort({col_ref, dir, null_pos}, limit, offset);
+}
+
+BuilderNode BuilderNode::sort(BuilderExpr col_ref,
+                              const std::string& dir,
+                              const std::string& null_pos,
+                              size_t limit,
+                              size_t offset) const {
+  return sort({col_ref, dir, null_pos}, limit, offset);
+}
+
+BuilderNode BuilderNode::sort(std::initializer_list<BuilderExpr> col_refs,
+                              SortDirection dir,
+                              NullSortedPosition null_pos,
+                              size_t limit,
+                              size_t offset) const {
+  return sort(toSortFields(col_refs, dir, null_pos), limit, offset);
+}
+
+BuilderNode BuilderNode::sort(std::initializer_list<BuilderExpr> col_refs,
+                              const std::string& dir,
+                              const std::string& null_pos,
+                              size_t limit,
+                              size_t offset) const {
+  return sort(toSortFields(col_refs, dir, null_pos), limit, offset);
+}
+
+BuilderNode BuilderNode::sort(const std::vector<BuilderExpr>& col_refs,
+                              SortDirection dir,
+                              NullSortedPosition null_pos,
+                              size_t limit,
+                              size_t offset) const {
+  return sort(toSortFields(col_refs, dir, null_pos), limit, offset);
+}
+
+BuilderNode BuilderNode::sort(const std::vector<BuilderExpr>& col_refs,
+                              const std::string& dir,
+                              const std::string& null_pos,
+                              size_t limit,
+                              size_t offset) const {
+  return sort(toSortFields(col_refs, dir, null_pos), limit, offset);
+}
+
+BuilderNode BuilderNode::sort(const BuilderSortField& field,
+                              size_t limit,
+                              size_t offset) const {
+  return sort(std::vector<BuilderSortField>({field}), limit, offset);
+}
+
+BuilderNode BuilderNode::sort(const std::vector<BuilderSortField>& fields,
+                              size_t limit,
+                              size_t offset) const {
+  std::vector<SortField> collation;
+  for (auto field : fields) {
```

Comment:
May cause a deep copy. Is this intentional?

Reply:
It's a bug. Thanks. Will fix.